interface/main/tabs/main.php

```diff
@@ -172,6 +175,42 @@ function isEncounterLocked( encounterId ) {
 ?>;
 // Set the csrf_token_js token that is used in the below js/tabs_view_model.js script
 var csrf_token_js = <?php echo js_escape(CsrfUtils::collectCsrfToken()); ?>;
+// will fullfill json and return promise if needed
+// to call elsewhere for a local scope ie
+// let localJson = top.jsFetchGlobals().then(data => {do something with parsed json data});
+// will use post here due to content type and length.
+function jsFetchGlobals(scope) {
+    let url = webroot_url + "/library/ajax/phpvars_to_js.php";
+    return new Promise((resolve, reject) => {
```

Comment:
whooohooo, a Promise. I don't really know what this is (gotta read up on it), but I think this is the first one in the codebase! whooohooooo

Reply:
Take a look at alertmsg function in dialog code as one use. In that case I want to ensure that the translations are available before I execute an alert. There are many areas in openemr where i've wanted to use but, in the end, too lazy to do the rewrites.:) Also you may want to look at using async awaits and deferred etc.
Another point is that I had this as a fetch where es6 the promise is implied and a simple return of the fetch call will return a promise. However, Edge v17html does not support(a Bug) nor IE11 so I was kind of pissed I couldn't use the way I wanted.  Then again, these days I get pissed if the car doesn't start by the third rotation. Cantankerous I believe it's called:)